message/messaging_service.cc

```diff
@@ -496,6 +498,8 @@ static constexpr unsigned do_get_rpc_client_idx(messaging_verb verb) {
     case messaging_verb::MUTATION_DONE:
     case messaging_verb::MUTATION_FAILED:
         return 3;
+    case messaging_verb::FORWARD_REQUEST:
+        return 4;
```

Comment:
```c++
    , _clients(2 + scfg.statement_tenants.size() * 2)
```

May need to be `* 3`.

Reply:
(not sure, need to see how _clients is sized)

Reply:
Did you check this?

Reply:
I agree with Avi, we introduced a per-tenant connection type, so it should be `* 3` there.

Other places in this file need checking, too. I can see that adjustment are needed in the following functions:

- `initial_scheduling_info`
- `get_rpc_client_idx`

...but maybe there are more.

It would be nice to have constants which represent the number of per-tenant and per-node connections and use them in the calculations instead of the magic number literals.